File: leetcode/204/204.go
package main

func countPrimes(n int) int {
	isPrime := make([]bool, n)
	prime := make([]int, 0)
	for i := range isPrime {
		isPrime[i] = true
	}

	for i := 2; i < n; i++ {
		if isPrime[i] {
			prime = append(prime, i)
		}
		for _, j := range prime {
			target := i * j
			if target >= n {
				break
			}
			isPrime[target] = false
			if (i % j) == 0 {
				break
			}
		}
	}

	return len(prime)
}

File: leetcode/204/204.py
class Solution:
    def countPrimes(self, n: int) -> int:
        prime = []
        isPrime = [True] * n

        for i in range(2, n): 
            if isPrime[i]: 
                prime.append(i)
            for j in prime:
                target = i * j
                if target >= n:
                    break
                isPrime[target] = False
                if (i % j) == 0:
                    break
        
        return len(prime)